From e1bd9182347c81732aa7fba181a0e71ff478fffd Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:03 +0800
Subject: [PATCH 038/150] dt-bindings: Add bindings for Phytium HW RNG

This patch documents the DT bindings for Phytium hardware random
number generator.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I60224653b154c072f10e37032dc148b2a1a3cfff
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/rng/phytium,rng.yaml  | 30 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 31 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/rng/phytium,rng.yaml

diff --git a/Documentation/devicetree/bindings/rng/phytium,rng.yaml b/Documentation/devicetree/bindings/rng/phytium,rng.yaml
new file mode 100644
index 000000000000..e32fc39ea86c
--- /dev/null
+++ b/Documentation/devicetree/bindings/rng/phytium,rng.yaml
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/rng/phytium,rng.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Random Number Generator
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,rng
+
+  reg:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    rng@0x31a06000 {
+        compatible = "phytium,rng";
+        reg = <0x0 0x31a06000 0x0 0x1000>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 41f9e4613397..602df4339a71 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2538,6 +2538,7 @@ F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
 F:	Documentation/devicetree/bindings/mtd/phytium,nfc.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
+F:	Documentation/devicetree/bindings/rng/phytium,rng.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,spi.yaml
-- 
2.47.0

